activemodel/lib/active_model/secure_password.rb

```diff
@@ -77,6 +77,29 @@ module ClassMethods
       #
       #   user.authenticate("vr00m")                                     # => false, old password
       #   user.authenticate("nohack4u")                                  # => user
+      #
+      # ===== Conditionally requiring a password
+      #
+      #   class Account
+      #     include ActiveModel::SecurePassword
+      #
+      #     attr_accessor :is_guest, :password_digest
+      #
+      #     has_secure_password
+      #
+      #     def errors
+      #       errors = super
+      #       errors.delete(:password, :blank) if is_guest
```

Comment:
I'd inline the exception with a tap, so: `super.tap { |errors| errors.delete(:password, :blank) if is_guest }`